impl/random/ch32v0xx.h

```diff
@@ -0,0 +1,100 @@
+/**********************************************************************************
+ * File Name          : ch32v0xx.h (Tentative)
+ * Author             : Charles Lee Scoville
+ * Version            : V0.0.2 (Pull request #153 modifications)
+ * Start Date         : 2024/01/02
+ * Description        : TRNG initialization shim
+ *
+ * The CH32V0XX does not have a hardware random number generator. This leaves us with
+ * only a few options for collecting entropy.
+ *
+ *  - Attach an entropy source circuit to the chip. (biased Zener, transistor, etc.)
+ *  - Use uninitialized data from SRAM proceeding chip power on.
+ *  - Timing difference (jitter) between two *independent* clock sources.
+ *  - Read some voltage with the ADC and look only at low bit(s).
+ *
+ *  *TODO* I intend to survey all options and make code for each, and a complete
+ *  write up. Until then, at least I have something working today. Tangible PoC
+ *  beats "good ideas" that haven't been written yet any day.
+ *
+ *  *TODO* Compilation flags to switch between the different entropy options
+ *  in a way that is transparent to this shim would be really neato.
+ *********************************************************************************/
+
+#include <ch32v00x.h>
+
+/* This must be defined in your CH32V003 project, and must be filled with raw /
+ * unfiltered samples from the internal reference voltage, as the name implies*/
+extern uint16_t ADC_Channel_Vrefint_value;
+
+
+uint32_t entropy_condenser_extractor(void) {
+
+	volatile uint64_t entropy_buffer = 0;
+	volatile uint8_t entropy_buffer_count = 0;
+
+	volatile uint32_t extract = 0;
+	volatile uint8_t extract_count = 0;
+	volatile FlagStatus extract_RDY = RESET;
+
+	if ( !((ADC1->CTLR2 & 1) != 0) )
+		while(1); /* ADC1 OFF!!! PANIK! */
```

Comment:
suggestion: dont block your app/system, make usage of errno*

errno = ENOXXX ;
return 0;


https://man7.org/linux/man-pages/man3/errno.3.html

Reply:
Good ask! +1
I'll see if I can get time to do that this weekend.

Reply:
It is done.... Took me long enough! 
(Had irl issues I had to clear up.)